### PR TITLE
Enable automerge on dependabot pull requests

### PR DESCRIPTION
**Pull request by feelepxyz**
Enable automerge on dependabot pull requests if the PR. This won't actually merge the PR, it will still require an approval from a team member.

### CODE REVIEW

File: .github/workflows/dependabot-automerge.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  contents: read
+  pull-requests: write
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.0.0
```
Comment (jurre) [inline .github/workflows/dependabot-automerge.yml:13]: What are we using the metadata for at this point?

Comment (feelepxyz) [reply .github/workflows/dependabot-automerge.yml:13]: Yeah it's a bit obscure but it ends up checking the PR is from dependabot, has only one commit and is verified

Comment (feelepxyz) [reply .github/workflows/dependabot-automerge.yml:13]: The action will error if those checks don't pass

File: .github/workflows/dependabot-automerge.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+on: pull_request_target
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
```
Comment (feelepxyz) [inline .github/workflows/dependabot-automerge.yml:11]:
```suggestion
      - name: Verify pull request is from Dependabot
```

File: .github/workflows/dependabot-automerge.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Verify pull request is from Dependabot
+        id: metadata
```
Comment (feelepxyz) [inline .github/workflows/dependabot-automerge.yml:12]:
```suggestion
```

**Review by jurre — APPROVED** [review]

**Pull request merged**